test/Microsoft.AspNetCore.Authentication.Core.Test/AuthenticationSchemeProviderTests.cs

```diff
@@ -117,6 +119,33 @@ public class AuthenticationSchemeProviderTests
             Assert.NotNull(await provider.GetDefaultSignOutSchemeAsync());
         }
 
+        [Fact]
+        public void SchemeRegistrationIsCaseSensitive()
+        {
+            var services = new ServiceCollection().AddOptions().AddAuthenticationCore(o =>
+            {
+                o.AddScheme<Handler>("signin", "whatever");
+                o.AddScheme<Handler>("signin", "whatever");
+            }).BuildServiceProvider();
+
+            Assert.Throws<InvalidOperationException>(() => services.GetRequiredService<IAuthenticationSchemeProvider>());
```

Comment:
Is this assertion really testing that scheme registration is case sensitive?

Reply:
Yes. It throws an `InvalidOperationException` with the text `"Scheme already exists: signin"` [here](https://github.com/aspnet/HttpAbstractions/blob/9f64c69a2a1005aed3b044209bbb228210d9a510/src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs#L127).

Reply:
Ok for tests like this its usually better to do something like this so its clear just from reading the code

```
var error = Assert.Throws<InvalidOperationException>(() => services.GetRequiredService<IAuthenticationSchemeProvider>());
Assert.Contains("Scheme already exists: signin", error.Message);
```

Reply:
Sure, if asserting on exception messages is your thing.